pkg/eval/str/str.go

```diff
@@ -8,6 +8,316 @@ import (
 	"github.com/elves/elvish/pkg/eval"
 )
 
+//elvdoc:fn compare
+//
+// ```elvish
+// str:compare $a $b
+// ```
+//
+// Compare returns an integer comparing two strings lexicographically. The
+// result will be 0 if a==b, -1 if a < b, and +1 if a > b.
+//
+// ```elvish-transcript
+// ~> str:compare a a
+// > 0
```

Comment:
Please stick to the default value output marker ▶.

Reply:
Done.